cpp/dolfinx/function/Function.h

```diff
@@ -147,7 +148,9 @@ class Function
   std::shared_ptr<const FunctionSpace> _function_space;
 
   // The vector of expansion coefficients (local)
-  la::PETScVector _vector;
+  std::shared_ptr<la::Vector<PetscScalar>> _x;
+
+  // The vector of expansion coefficients
+  mutable Vec _vector = nullptr;
```

Comment:
Vector here being stored twice is weird. Is `_vector` only for caching? Is this duplicating the storage?

Reply:
`Vec` is a PETSc vector that wraps the data in `_x`. There is no duplication of storage. `_vector` will probably move at some point once the design becomes clearer, but for now this avoids user-level changes.